spring-pulsar-docs/src/main/antora/modules/ROOT/pages/reference/topic-resolution.adoc

```diff
@@ -34,6 +34,22 @@ NOTE: The `message-type` is the fully-qualified name of the message class.
 
 WARNING: If the message (or the first message of a `Publisher` input) is `null`, the framework won't be able to determine the topic from it. Another method shall be used to specify the topic if your application is likely to send `null` messages.
 
+=== Specified via annotation
+
+When no topic is passed into the API and there are no custom topic mappings configured, the system looks for a `@PulsarTypeMapping` annotation on the class of the message being produced or consumed.
+The default topic can be specified via the `topic` attribute on the annotation.
+
+The following example configures the default topic to use when producing or consuming messages of type `Foo`:
+
+[source,java,indent=0,subs="verbatim"]
+----
+@PulsarTypeMapping(topic = "foo-topic")
+record Foo(String value) {
+}
+----
+
+NOTE: The annotations are looked up on-demand and their result is cached. However, there is still a small performance hit on the first lookup. If you want to disable this feature you can invoke the `usePulsarTypeMappingAnnotations(false)` method on the `DefaultTopicResolver`.
```

Comment:
Do we really need this? If you do not want your type and/or topic information to be resolved because of performance reason why would you in the first place have opted to use the `@PulsarTypeMapping` annotation?

Reply:
It is more of an "emergency handle" to allow users to disable if they need to. It is not exposed via properties. In that spirit of not really advertising it I am fine w/ removing this here as well. If a bug comes up and someone needs this we can easily point them to it. I will remove this in both places.

Reply:
Done - PTAL